tethys_apps/base/url_map.py

```diff
@@ -57,14 +61,16 @@ def __init__(
         self.custom_match_regex = regex
         self.handler = handler
         self.handler_type = handler_type
+        self.title = title
+        self.index = index
 
     def __repr__(self):
         """
         String representation
         """
         return (
             f"<UrlMap: name={self.name}, url={self.url}, controller={self.controller}, protocol={self.protocol}, "
-            f"handler={self.handler}, handler_type={self.handler_type}>"
+            f"handler={self.handler}, handler_type={self.handler_type}, title={self.title}, index={self.index}>"
```

Comment:
Is this only being used by the ReactPy stuff, or is it/can it be leveraged more broadly?

Reply:
This is only being used for ReactPy stuff. I did not put any though into whether it could be used more broadly. This was implemented specifically for use in the automatic generation of the navigation bar links - one per function decorated by `@page`.